modules/login/login_module.e

```diff
@@ -0,0 +1,1037 @@
+				create lnk.make ("Login", "account/roc-login")
+			end
+			a_menu_system.primary_menu.extend (lnk)
+			lnk.set_weight (98)
+		end
+
+	block_list: ITERABLE [like {CMS_BLOCK}.name]
+		local
+			l_string: STRING
+		do
+			Result := <<"login","register","reactivate","new_password", "reset_password">>
+			create l_string.make_empty
+			across Result as ic loop
+					l_string.append (ic.item)
+					l_string.append_character (' ')
+				end
+			write_debug_log (generator + ".block_list:" + l_string )
+		end
+
+	get_block_view (a_block_id: READABLE_STRING_8; a_response: CMS_RESPONSE)
+		local
+			vals: CMS_VALUE_TABLE
+		do
+			if
+				a_block_id.is_case_insensitive_equal_general ("login") and then
+				a_response.request.path_info.starts_with ("/account/roc-login")
+			then
+				get_block_view_login (a_block_id, a_response)
+			elseif
+				a_block_id.is_case_insensitive_equal_general ("register") and then
+				a_response.request.path_info.starts_with ("/account/roc-register")
+			then
+				get_block_view_register (a_block_id, a_response)
+			elseif
+				a_block_id.is_case_insensitive_equal_general ("reactivate") and then
+				a_response.request.path_info.starts_with ("/account/reactivate")
+			then
+				get_block_view_reactivate (a_block_id, a_response)
+			elseif
+				a_block_id.is_case_insensitive_equal_general ("new_password") and then
+				a_response.request.path_info.starts_with ("/account/new-password")
+			then
+				get_block_view_new_password (a_block_id, a_response)
+			elseif
+				a_block_id.is_case_insensitive_equal_general ("reset_password") and then
+				a_response.request.path_info.starts_with ("/account/reset-password")
+			then
+				get_block_view_reset_password (a_block_id, a_response)
+			end
+		end
+
+	handle_login (api: CMS_API; req: WSF_REQUEST; res: WSF_RESPONSE)
+		local
+			r: CMS_RESPONSE
+			br: BAD_REQUEST_ERROR_CMS_RESPONSE
+		do
+			create {GENERIC_VIEW_CMS_RESPONSE} r.make (req, res, api)
+			r.set_value ("Login", "optional_content_type")
+			r.execute
+		end
+
+	handle_workaround_filter (api: CMS_API; req: WSF_REQUEST; res: WSF_RESPONSE)
+		local
+			r: CMS_RESPONSE
+			br: BAD_REQUEST_ERROR_CMS_RESPONSE
+		do
+			create {GENERIC_VIEW_CMS_RESPONSE} r.make (req, res, api)
+			r.execute
+		end
+
+
+	handle_logout (api: CMS_API; req: WSF_REQUEST; res: WSF_RESPONSE)
+		local
+			r: CMS_RESPONSE
+			l_url: STRING
+			l_oauth_gmail: OAUTH_LOGIN
+			l_cookie: WSF_COOKIE
+		do
+			if
+				attached {WSF_STRING} req.cookie ({LOGIN_CONSTANTS}.oauth_session) as l_cookie_token and then
+				attached {CMS_USER} current_user (req) as l_user
+			then
+					-- Logout gmail
+				create l_cookie.make ({LOGIN_CONSTANTS}.oauth_session, l_cookie_token.value)
+				l_cookie.set_path ("/")
+				l_cookie.set_max_age (-1)
+				res.add_cookie (l_cookie)
+				unset_current_user (req)
+				create {GENERIC_VIEW_CMS_RESPONSE} r.make (req, res, api)
+				r.set_status_code ({HTTP_CONSTANTS}.found)
+				r.set_redirection (req.absolute_script_url (""))
```

Comment:
probably, we should add a 
{CMS_RESPONSE}.front_page_url: STRING

by default this would be the "" ... but could be "/welcome" if dev decides so